join.go

```diff
@@ -207,11 +206,6 @@ func (n *JoinNode) matchPoints(p srcPoint) {
 				break
 			}
 		}
-		if n.allReported {
-			// Can't trust lowMark until all parents have reported.
-			// Remove any unneeded match points.
-			n.matchGroupsBuffer[groupId] = matches[i:]
-		}
```

Comment:
This block above,  looks like it creates duplicates when stuff hasn't reported.  I removed it, it could be related to an issue that has been seen.